### PR TITLE
Initial Compose ReaderScreen

### PR DESCRIPTION
**Pull request by VDostoyevskiy**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/CommonOperations.kt [hunk 1]
```diff
@@ -433,3 +446,32 @@ suspend fun DialogState.showDatePicker(): String? {
...
+    t2: T2,
+    t3: T3,
+    block: context(T1, T2, T3)
+    () -> R,
+): R {
+    contract {
+        callsInPlace(block, EXACTLY_ONCE)
+    }
+    return block(t1, t2, t3)
+}
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/CommonOperations.kt:477]: nit: Move those to somewhere else.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import eu.kanade.tachiyomi.source.model.Page.State.LOAD_PAGE
+import eu.kanade.tachiyomi.source.model.Page.State.QUEUE
+import eu.kanade.tachiyomi.source.model.Page.State.READY
+
+@Destination
+@Composable
+fun ReaderScreen(info: BaseGalleryInfo, page: Int = -1) {
+    LockDrawer(true)
+    val pageLoader = remember {
+        val dir = DownloadManager.getDownloadInfo(info.gid)?.archiveFile
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:48]: nit: Naming.

**Review by FooIbar — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/CommonOperations.kt (hunk 1 above)
Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/CommonOperations.kt]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt (hunk 2 above)
Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:48]: Done

**Pull request merged**